Tools/AP_Periph/esc_apd_telem.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+  ESC Telemetry for Hobbywing Pro 80A HV ESC. This will be
+  incorporated into a broader ESC telemetry library in ArduPilot
+  master in the future
+
+  This protocol only allows for one ESC per UART RX line, so using a
+  CAN node per ESC works well.
+ */
+#include "esc_apd_telem.h"
+#include <AP_HAL/utility/sparse-endian.h>
+#include <AP_Math/crc.h>
+#include <AP_Math/definitions.h>
+#include <string.h>
+
+#ifdef HAL_PERIPH_ENABLE_ESC_APD
+
```

Comment:
When following a public protocol, please post a link to the protocol at the top of the file for reference. I see you posted it down below in the temperature area but I was digging around for it before I saw it there.

https://docs.powerdrives.net/products/hv_pro/uart-telemetry-output